lib/errorHandlers/index.ts

```diff
@@ -7,37 +6,39 @@ const {
   isValidationError,
   isMissingScopeError,
 } = require('@hubspot/local-dev-lib/errors/index');
+import { FileSystemError } from '@hubspot/local-dev-lib/models/FileSystemError';
+import { HubSpotHttpError } from '@hubspot/local-dev-lib/models/HubSpotHttpError';
+import { BaseError } from '@hubspot/local-dev-lib/types/Error';
 const { shouldSuppressError } = require('./suppressError');
 const { i18n } = require('../lang');
 const util = require('util');
-const { isAxiosError } = require('axios');
 
 const i18nKey = 'lib.errorHandlers.index';
 
-function logError(error, context = {}) {
+export function logError(error: unknown, context?: typeof ApiErrorContext) {
   debugError(error, context);
 
   if (
     shouldSuppressError(error, context) ||
-    shouldSuppressError(error, error.context)
+    shouldSuppressError(error, (error as HubSpotHttpError).context)
   ) {
     return;
   }
 
-  if (isHubSpotHttpError(error) && context) {
-    error.updateContext(context);
+  if (isHubSpotHttpError(error) && context instanceof ApiErrorContext) {
+    (error as HubSpotHttpError).updateContext(context);
   }
 
   if (isHubSpotHttpError(error) || isFileSystemError(error)) {
     if (isValidationError(error) || isMissingScopeError(error)) {
-      logger.error(error.formattedValidationErrors());
+      logger.error((error as HubSpotHttpError).formattedValidationErrors());
     } else {
-      logger.error(error.message);
+      logger.error((error as HubSpotHttpError | FileSystemError).message);
     }
   } else if (isSystemError(error)) {
-    logger.error(error.message);
-  } else if (error.message || error.reason) {
-    const message = [];
+    logger.error((error as BaseError).message);
```

Comment:
`isSystemError` should be type narrowing, is this assertion necessary?

Reply:
Addressed in https://github.com/HubSpot/hubspot-cli/pull/1251/commits/9fc06a16fccfda6b8e5d060b6c4703e0a83a0d07